trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/shared/component/StatePlaceholder.kt

```diff
@@ -0,0 +1,90 @@
+package net.thechance.mena.trends.presentation.shared.component
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import mena.trends_presentation.generated.resources.Res
+import mena.trends_presentation.generated.resources.ic_no_connection
+import mena.trends_presentation.generated.resources.re_try
+import net.thechance.mena.designsystem.presentation.component.button.PrimaryButton
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.theme.theme.MenaTheme
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+
+@Composable
+fun StatePlaceholder(
+    modifier: Modifier = Modifier,
+    isBottomVisible: Boolean = false,
+    stateIcon: Painter,
+    stateTitle: String,
+    stateDescription: String,
+    onRetry: () -> Unit = {}
```

Comment:
i think this order for parameter is better
required
optional
call back 

```suggestion
    stateIcon: Painter,
    stateTitle: String,
    stateDescription: String,
    modifier: Modifier = Modifier,
    isBottomVisible: Boolean = false,
    onRetry: () -> Unit = {}